frontend/src/templates/Field/ChildrenCompound/ChildrenCompoundField.tsx

```diff
@@ -271,6 +271,14 @@ const ChildrenBody = ({
       ) {
         return ''
       }
+      // We use the childname to indicate if there is a child below 21.
+      // If the childname is an empty string, it means there is no child below 21.
+      // As our definition of child in FormSG means child below 21, we want to
+      // return empty strings for other child attributes even if their value is populated by myinfo
+      // if there is no childname.
+      if (myInfoChildrenBirthRecords.childname?.[0] === '') {
```

Comment:
Why `0` and not `indexOfChild`? the array index at the end is for the specific child right? Or am I not getting the shape of the myInfoChildren correct 😅

Reply:
Yup you're right. Previously I'd only considered the case where a parent only had children above 21yo (so the first childname would definitely be `''`), but you're right that this should be according to `indexOfChild` because this assumption might not be true. Made the change in [e35bc74](https://github.com/opengovsg/FormSG/pull/6715/commits/e35bc74e517f0eb7a00dc697ceda07bd3b5e35a1).

When checking another case (1 child below 21, 1 child above 21), I realised the one above 21 still showed up in the child names selection as a blank dropdown, so I added a `useMemo` for `childNameValues` in that commit as well!

Reply:
Thanks for the additional considerations @wanlingt, man do we have a way to test this HAHAHAHA are there anyone with kids above and below 21 😂